iina/PrefKeyBindingViewController.swift

```diff
@@ -155,101 +144,74 @@ class PrefKeyBindingViewController: NSViewController, PreferenceWindowEmbeddable
-      let fm = FileManager.default
-      // - if exists
-      if fm.fileExists(atPath: newFilePath) {
-        if Utility.quickAskPanel("config.file_existing", sheetWindow: self.view.window) {
-          // - delete file
-          do {
-            try fm.removeItem(atPath: newFilePath)
-          } catch {
-            Utility.showAlert("error_deleting_file", sheetWindow: self.view.window)
-            return
-          }
-        } else {
-          NSWorkspace.shared.activateFileViewerSelecting([URL(fileURLWithPath: newFilePath)])
-          return
-        }
-      }
-      // - new file
+      guard let newFilePath = self.checkNewConfigFile(with: newName) else { return }
+
       if !fm.createFile(atPath: newFilePath, contents: nil, attributes: nil) {
         Utility.showAlert("config.cannot_create", sheetWindow: self.view.window)
         return
       }
-      // save
-      self.userConfigs[newName] = newFilePath
-      Preference.set(self.userConfigs, for: .inputConfigs)
-      // load
-      self.currentConfName = newName
-      self.currentConfFilePath = newFilePath
-      self.userConfigNames.append(newName)
-      self.confTableView.reloadData()
-      self.confTableSelectRow(withTitle: newName)
-      self.loadConfigFile()
+      self.loadConfigFile(newName)
     }
   }
 
-
   @IBAction func duplicateConfFileAction(_ sender: AnyObject) {
-    // prompt
     Utility.quickPromptPanel("config.duplicate", sheetWindow: view.window) { newName in
-      if self.userConfigs[newName] != nil || PrefKeyBindingViewController.defaultConfigs[newName] != nil {
-        Utility.showAlert("config.name_existing", sheetWindow: self.view.window)
+      guard let newFilePath = self.checkNewConfigFile(with: newName) else { return }
+
+      do {
+        try fm.copyItem(atPath: self.currentConfFilePath!, toPath: newFilePath)
+      } catch let error {
+        Utility.showAlert("config.cannot_create", arguments: [error.localizedDescription], sheetWindow: self.view.window)
         return
       }
-      // copy
-      let currFilePath = self.currentConfFilePath!
-      let newFileName = newName + ".conf"
-      let newFilePath = Utility.userInputConfDirURL.appendingPathComponent(newFileName).path
-      let fm = FileManager.default
-      // - if exists
-      if fm.fileExists(atPath: newFilePath) {
-        if Utility.quickAskPanel("config.file_existing", sheetWindow: self.view.window) {
-          // - delete file
-          do {
-            try fm.removeItem(atPath: newFilePath)
-          } catch {
-            Utility.showAlert("error_deleting_file", sheetWindow: self.view.window)
-            return
-          }
-        } else {
-          NSWorkspace.shared.activateFileViewerSelecting([URL(fileURLWithPath: newFilePath)])
-          return
-        }
-      }
-      // - copy file
+      self.loadConfigFile(newName)
+    }
+    
+  }
+  
+  @IBAction func configFileListDoubleAction(_ sender: NSTableView) {
```

Comment:
TBH I think when double clicking a table row people should expect the cell become editable, rather than a pop-up dialog. It's fine though.

Reply:
Initially I wanted to implement this way, however, I noticed that if the renaming fails, it would be messy to bring up another dialog window to give the user info on why the action failed. So I just used the alert window as other actions (new, duplicate) in this window.